src/pages/settings/GenericSettings.tsx

```diff
@@ -148,7 +149,16 @@ export function GenericSettings({
                                                     !isTouchscreenDevice &&
                                                     typeof page === "undefined")
                                             }
-                                            onClick={() => switchPage(entry.id)}
+                                            onClick={() =>
+                                                !entry.disabled &&
+                                                switchPage(entry.id)
+                                            }
+                                            disabled={entry.disabled}
+                                            style={
+                                                entry.disabled
+                                                    ? "filter: grayscale(70%) brightness(70%);"
+                                                    : undefined
+                                            }
```

Comment:
These inline styles should probably just be implemented in ButtonItem.

Reply:
like this?

(missed this thanks to github not sending me a notif)